docs/source/tutorials_source/package/tutorial_custom_augmentations.py

```diff
@@ -0,0 +1,366 @@
+# we create a new dataloader. This time, we can pass the transform directly to the dataset.
+
+# test transforms differ from training transforms as they do not introduce
+# additional noise
+test_transforms = torchvision.transforms.Compose([
+    HistogramNormalize(),
+    torchvision.transforms.Grayscale(num_output_channels=3),
+    torchvision.transforms.Resize(input_size),
+    torchvision.transforms.ToTensor(),
+])
+
+# create the dataset and overwrite the image loader as before
+dataset_test = lightly.data.LightlyDataset(
+    input_dir=path_to_data,
+    transform=test_transforms
+)
+dataset_test.dataset.loader = tiff_loader
+
+# create the test dataloader
+dataloader_test = torch.utils.data.DataLoader(
+    dataset_test,
+    batch_size=1,
+    shuffle=False,
+    drop_last=False,
+    num_workers=num_workers
+)
+
+# get vector representations for all images in the dataset
+encoder.to('cuda')
+embeddings, _, fnames = encoder.embed(dataloader_test, device='cuda')
+embeddings = normalize(embeddings)
+
+# %%
+# Now, we can use the embeddings to search for nearest neighbors.
+#
+# We choose three example images. For each example image, we find 50 nearest neighbors.
+# Then, we plot the critical findings in the example image (dark blue) and the distribution
+# of the critical findings in the nearest neighbor images (light blue) as bar plots.
+
+
+# transform the original annotations to multiclass labels
+fnames = [fname.split('.')[0] for fname in fnames]
+
+df = pandas.read_csv('/datasets/vinbigdata/train.csv')
+classes = list(np.unique(df.class_name))
+filenames = list(np.unique(df.image_id))
+
+multilabels = np.zeros((len(dataset_test.get_filenames()), len(classes)))
+for filename, label in zip(df.image_id, df.class_name):
+    try:
+        i = fnames.index(filename.split('.')[0])
+        j = classes.index(label)
+        multilabels[i, j] = 1.
+    except:
+        pass
+
+
+def plot_knn_multilabels(embeddings, multilabels, samples_idx, n_neighbors=50):
+    """Plots multiple rows of random images with their nearest neighbors
+    """
+    # lets look at the nearest neighbors for some samples
+    # we use the sklearn library
+    nbrs = NearestNeighbors(n_neighbors=n_neighbors).fit(embeddings)
+    _, indices = nbrs.kneighbors(embeddings)
+
+    # TODO
+    bar_width = 0.4
+    r1 = np.arange(multilabels.shape[1])
+    r2 = r1 + bar_width
+
+    # loop through our randomly picked samples
+    for idx in samples_idx:
+        fig = plt.figure()
+    
+        bars1 = multilabels[idx]
+        bars2 = np.mean(multilabels[indices[idx]], axis=0)
+
+        plt.bar(r1, bars1, color='steelblue', edgecolor='black', width=bar_width)
+        plt.bar(r2, bars2, color='lightsteelblue', edgecolor='black', width=bar_width)
+        plt.xticks(0.5 * (r1 + r2), classes, rotation=90)
+        plt.tight_layout()
+
+
+plot_knn_multilabels(embeddings, multilabels, [4111, 3340, 1796])
```

Comment:
Add more information to the plot. Why do we have this plot (like you did in the presentation).